src/controller/CHIPDeviceControllerSystemState.h

```diff
@@ -106,19 +106,25 @@ class DeviceControllerSystemState
     using CASEClientPool        = DeviceControllerSystemStateParams::CASEClientPool;
 
 public:
-    ~DeviceControllerSystemState(){};
+    ~DeviceControllerSystemState()
+    {
+        VerifyOrDie(mRefCount == 0);
+        VerifyOrDie(mWasShutdown);
```

Comment:
This `mWasShutdown` assert is very much failing in the Darwin tests.

Reply:
Can you paste the log? I've been looking for it, and I cannot find it.

Reply:
Those logs are gone now that this PR got updated, sorry... :(